faq.md

```diff
@@ -0,0 +1,99 @@
+
+## How do I use the Sanitizer API
+
+Construct a new Sanitizer, using either the defaults or suppliying a custom
+configuration, and then pass in a string, a fragment, or an entire document
+for sanitization.
+
+```js
+const s = new Sanitizer();
+
+// Returns a DOM Fragment equivalent to "<em>Hello World</em>".
+s.sanitize("<em>Hello World</em><script>alert('surprise');</script>");
+```
+
+## Can I use the Sanitizer in my app?
+
+Not yet. Current browser support is preliminary and incomplete and still
+diverges in several details. And the specification itself is still in flux.
+We expect to be working through these issues in the coming months.
+
+If you wish to try it out already:
+
+* Firefox: Enable the `sanitizer.enabled` preference.
+* Chromium / Chrome: Start the browser with the
+  `--enable-blink-features=SanitizerAPI` flag.
+
+## Do I have to supply my own Sanitizer configuration? What is the default?
+
+The default configuration will block all content that causes script execution.
+unknown elements, plus a handful of deprecated elements.
+It is safe against any direct XSS.
+
+```js
+const default_sanitizer = new Sanitizer();
+
+// Returns a DOM Fragment equivalent to "<em>Hello World</em>".
+default_sanitizer.sanitize("<em>Hello World</em>");
+
+// Complex formatting is allowed by default. This will pass unchanged.
+default_sanitizer.sanitize("<div id='hello'><span style='font-weight: bold'><Hello World</span></div>");
+
+// Script execution is no-no: Returns "<p>Hello</p>"
+default_sanitizer.sanitize("<p onclick='alert(\'surprise\')'>Hello</p>");
+```
+
+## Is Sanitizer output secure?
+
+'Secure' means different things in different contexts. The Sanitizer guarantees
+safety from direct script execution in all configurations. This cannot be
+overridden by custom configurations.
+
+```js
+// Produces an empty fragment. There is no way to allow the <script> element.
+new Sanitizer({"allowElements": ["script"]}.sanitize("<script>alert('surprise')</script>");
+```
+
+Other notions of safety can likely be configured. For example, if an
+application relies on the `id=` attribute and does not wish user-supplied input
+to introduce unforseen `id=` attribute values, then the application can easily
+configure a block a Sanitizer to block the this attribute. However, the
+application must ensure on its own that all parts of the application
+(including any of its dependencies) will adhere to this restriction.
+
+
+## When should I use `.sanitize`, when `.sanitizeToString`?
+
+This is up to you. `.sanitize` and `.sanitizeToString` do the same work,
+except `.sanitizeToString` will serialize the result fragment into a string
+and return that. Since in all likelihood the Sanitizer result will eventually
+be inserted into the DOM, retaining the fragment (i.e. `.sanitize`) rather
+than a serialization that needs to be re-parsed (i.e. `.sanitizeToString`)
+is more efficient.
+
+Additionally, handling a string opens up the possibility of mXSS if the
```

Comment:
Maybe link "mXSS" off to something like https://hackinparis.com/data/slides/2013/slidesmarioheiderich.pdf?

Reply:
Ah, you had a TODO below. :)

Reply:
I'd nudge towards sanitize over the string-variant more intensively. Specifically, I wouldn't start with "This is up to you".

How about:
```
We're recommending everyone to use `sanitize()` if possible:
* There are XSS risks with parsing & reparsing HTML strings, which are known as mXSS [link].
* In order for the Sanitizer to work, we're already parsing the supplied input into a DocumentFragment and analyze the DOM tree to identify XSS risk.. This DOM tree can be easily applied to an existing Node in your document. Turning it into a string only to just parse it again within the implementation of innerHTML carries a performance cost, that we can help you avoid.```

Reply:
Hmm. Generally, I'd like to stay in _"I'm-a-sign-not-a-cop"_ territory. I've reworded this along your suggestion, but kept it somewhat more explanatory and less prescriptive. Care to have another look?

Reply:
Link: Done.